### DISCUSSION TITLE
Automation testing with multiple viewports

### DISCUSSION THREAD

**Discussion by venugopalkathavate** [storybookjs/storybook#24942] [Help] [answered]
### Summary

With the test-runner, we can automatically run tests for all the stories with or without the play function. However, I was unable to find documentation or a solution on how to test each story in multiple viewports with the test-runner. Is it possible?

**Comment by yannbf** [comment, marked as answer]
Hey @venugopalkathavate!

The test-runner allows you to configure a specific snapshot per story, as can be seen in [this recipe](https://github.com/storybookjs/test-runner#preconfiguring-viewport-size). However, It does not have a way to multiply the same test with multiple configurations. If you really want to achieve that with the test-runner, you could potentially duplicate your stories and set a different viewport for each of these stories' parameters.

Alternatively, if you ever looked at Chromatic, they have a feature called story modes which does exactly what you want to achieve:
https://www.chromatic.com/docs/modes/#modes-for-testing-themes-viewports-locales-amp-more

**Comment by jussibility** [comment]
I would like to do the same except running snapshot tests for all themes. Is there no other way except duplicating stories if I don't want to use Chromatic?

**Reply by yannbf** [reply]
You could potentially render a single story multiple times, in different themes:
https://yannbraga.dev/blog/multi-theme-decorator#seeing-multiple-themes-at-the-same-time
